src/vm/arm64/asmhelpers.S

```diff
@@ -310,29 +310,46 @@ LOCAL_LABEL(shadowupdateend):
     ldp  x12, x13, [sp],#16
 #endif
 
-    // Branch to Exit if the reference is not in the Gen0 heap
+    // Branch to DoneCardTable if the reference is not in the Gen0 heap
```

Comment:
The problem was the card table needed the x15 register value to determine whether the reference was in the ephemeral heap.  The WW code needed to trash two registers.  

Since there are no barriers, between the card table and the WW, there is no guarantee which order they are observable by another thread (unless an exception occurs).  So I suspected I could reverse the order to solve the register problem.

If we have to reverse the order, I will need another scratch register, spill to the stack, drop the ephemeral heap check (which is done in AMD64 GC Server mode), or get creative...

I will assume the order is important and try to come up with a solution....